portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/BasePortalHandler.java

```diff
@@ -209,4 +210,14 @@ protected void addLocale(PortalRenderContext rcontext, Site site, String userId)
 			timeService.clearLocalTimeZone(userId);
 		}
 	}
+
+	protected void addServerTime(PortalRenderContext rcontext) {
+		rcontext.put("serverTime", Instant.now().toEpochMilli());
+	}
+
+	protected void addUserTimezoneOffset(PortalRenderContext rcontext) {
+
+		int offset = timeService.getLocalTimeZone().getOffset(Instant.now().toEpochMilli());
+		rcontext.put("userTimezoneOffset", offset);
+	}
```

Comment:
while we are here, why don't we add the user's current (localized) date/time as ISO-8601 .... and might as well add their preferred timezone (e.g., Europe/Stockholm) .......